server/src/definitions/DefinitionUtils.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Find in a text the last substring that could be a valid YML entity name's substring.
+ * @param text Some text.
+ * @returns the canditate entity's name or null if nothing could be found.
+ */
+export function getDocumentLastWord(text: string) {
```

Comment:
Confusing name since you don't really use it on whole “documents” and look for YML IDs rather than real words.